src/multi-jvm/scala/akka/contrib/datareplication/ReplicatorSpec.scala

```diff
@@ -481,5 +481,32 @@ class ReplicatorSpec extends MultiNodeSpec(ReplicatorSpec) with STMultiNodeSpec
     enterBarrier("after-8")
   }
 
+  "check that a gossip update and a local update both cause a change event to emit with the merged data" in {
+    val changedProbe = TestProbe()
+
+    runOn(second) {
+      replicator ! Subscribe("H", changedProbe.ref)
+      replicator ! Update("H", LWWMap.empty[Int], writeTwo)(_ + ("a" -> 1))
+      changedProbe.expectMsgPF() { case Changed("H", h: LWWMap[Int] @unchecked) ⇒ h.entries } should be(Map("a" -> 1))
+    }
+
+    enterBarrier("update-h1")
+
+    runOn(first) {
+      replicator ! Update("H", LWWMap.empty[Int], writeTwo)(_ + ("a" -> 2))
+    }
+
+    runOn(second) {
+      replicator ! Update("H", LWWMap.empty[Int], writeTwo)(_ + ("b" -> 2))
+      changedProbe.expectMsgPF() {
+        case Changed("H", h: LWWMap[Int] @unchecked) if h.entries.size == 1 ⇒
+          changedProbe.expectMsgPF() { case Changed("H", h: LWWMap[Int] @unchecked) ⇒ h.entries }
+        case Changed("H", h: LWWMap[Int] @unchecked) ⇒
+          h.entries
+      } should be(Map("a" -> 2, "b" -> 2))
```

Comment:
if there are clock differences between second and first you might end up with `"a" -> 1`, please use something else than LWWMap